packages/nextjs/src/appWithKeycloak.js

```diff
@@ -23,19 +23,25 @@ const appWithKeycloak = (
       : {}
   }
 
+  async function getNextAppInitialProps(appContext){
+    return NextApp.getInitialProps ? await NextApp.getInitialProps(appContext) : {};
+  }
+
   class AppWithKeycloak extends React.Component {
     static async getInitialProps(appContext) {
       const { isAuthenticated } = checkIfUserAuthenticated(appContext)
-      const [cmpInitialProps, keycloakInitConfig] = await Promise.all([
+      const [cmpInitialProps, keycloakInitConfig,nextAppProps] = await Promise.all([
         getComponentInitialProps(appContext),
-        getKeycloakInitialProps(appContext)
+        getKeycloakInitialProps(appContext),
+        getNextAppInitialProps(appContext)
       ])
 
       return {
         pageProps: {
           ...cmpInitialProps,
           isAuthenticated
         },
+        nextAppProps,
```

Comment:
@mohux ,

My primary concern is how this props can be used, with the changes applied you are calling the `NextApp.getInitialProps` method but the result is never used.

How should this props be exposed to the calling `App`? At the moment they are just getting ignored and I don't think the example you posted would work after applying the changes because of what I stated above.

Reply:
its optional, they will be empty if there is no getInitialProps, we can set them to null or never add them to return object

some people use getInitialProps inside app and need values for application level scope

for example

some people call an api inside App.js or retreive some data from server side across the whole project

if you try and use getInitialProps inside app.js 

you will see that it wont work at all

Reply:
I am also experiencing some problems with getInitialProps on the App level.
however, I think, the only fix required, is to call `NextApp.getInitialProps(appContext) ` instead of `appContext.Component.getInitialProps(appContext.ctx)` because the later is for end-users.